flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/DefaultSlotPoolServiceSchedulerFactory.java

```diff
@@ -224,20 +210,52 @@ public static DefaultSlotPoolServiceSchedulerFactory fromConfiguration(
 
     private static JobManagerOptions.SchedulerType getSchedulerType(
             Configuration configuration, JobType jobType, boolean isDynamicGraph) {
-        if (configuration.get(JobManagerOptions.SCHEDULER_MODE)
-                == SchedulerExecutionMode.REACTIVE) {
-            return JobManagerOptions.SchedulerType.Adaptive;
+        JobManagerOptions.SchedulerType schedulerType;
+        if (jobType == JobType.BATCH) {
+            if (configuration.get(JobManagerOptions.SCHEDULER_MODE)
+                            == SchedulerExecutionMode.REACTIVE
+                    || configuration.get(JobManagerOptions.SCHEDULER)
+                            == JobManagerOptions.SchedulerType.Adaptive) {
+                LOG.info(
+                        "Adaptive Scheduler configured, but Batch job detected. Changing scheduler type to 'AdaptiveBatch'.");
+                // overwrite
+                schedulerType = JobManagerOptions.SchedulerType.AdaptiveBatch;
+            } else {
+                schedulerType =
+                        configuration
+                                .getOptional(JobManagerOptions.SCHEDULER)
+                                .orElse(
+                                        isDynamicGraph
+                                                ? JobManagerOptions.SchedulerType.AdaptiveBatch
+                                                : JobManagerOptions.SchedulerType.Default);
+            }
```

Comment:
```suggestion
            
                schedulerType =
                        configuration
                                .getOptional(JobManagerOptions.SCHEDULER)
                                .orElse(
                                        isDynamicGraph
                                                ? JobManagerOptions.SchedulerType.AdaptiveBatch
                                                : JobManagerOptions.SchedulerType.Default);
                
                if (schedulerType = JobManagerOptions.SchedulerType.Adaptive) {
                    LOG.info(
                            "Adaptive Scheduler configured, but Batch job detected. Changing scheduler type to 'AdaptiveBatch'.");
                    schedulerType = JobManagerOptions.SchedulerType.AdaptiveBatch;
                }
```

Reply:
I think it's not needed to print that log if reactive mode is enabled for a batch job.